api/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/admin/controllers/DataLocationController.java

```diff
@@ -0,0 +1,93 @@
+package ca.bc.gov.tno.areas.admin.controllers;
+
+import java.util.List;
+
+import javax.annotation.security.RolesAllowed;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import ca.bc.gov.tno.dal.db.services.interfaces.IDataLocationService;
+import ca.bc.gov.tno.dal.db.entities.DataLocation;
+
+/**
+ * Endpoints to communicate with the TNO DB data data locations.
+ */
+@RolesAllowed("administrator")
+@RestController("AdminDataLocationController")
+@RequestMapping({ "/admin/data/locations", "/api/admin/data/locations" })
+public class DataLocationController {
```

Comment:
Data locations allow us to filter out data locations services don't currently support (i.e. NAS).